packages/flutter/test/cupertino/popup_surface_test.dart

```diff
@@ -0,0 +1,124 @@
+// found in the LICENSE file.
+
+// This file is run as part of a reduced test set in CI on Mac and Windows
+// machines.
+@Tags(<String>['reduced-test-set'])
+library;
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('light painted appearance', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      CupertinoApp(
+        theme: const CupertinoThemeData(brightness: Brightness.light),
+        home: buildRainbowPopupSurface(isSurfacePainted: true),
+      ),
+    );
+
+    // Golden verifies the backdrop filter effect of the CupertinoPopupSurface
+    // in dark mode when the popup surface is painted.
+    await expectLater(
+      find.byType(CupertinoApp),
+      matchesGoldenFile('cupertinoPopupSurface.light.filter-effects.png'),
+    );
+  });
+
+  testWidgets('dark painted appearance', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      CupertinoApp(
+        theme: const CupertinoThemeData(brightness: Brightness.dark),
+        home: buildRainbowPopupSurface(isSurfacePainted: true),
+      ),
+    );
+
+    // Golden matches the backdrop filter effect of the CupertinoPopupSurface
+    // in dark mode when the popup surface is painted.
+    await expectLater(
+      find.byType(CupertinoApp),
+      matchesGoldenFile('cupertinoPopupSurface.dark.filter-effects.png'),
+    );
+  });
+
+  testWidgets('unpainted appearance', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      CupertinoApp(
+        theme: const CupertinoThemeData(brightness: Brightness.light),
+        home: buildRainbowPopupSurface(isSurfacePainted: false),
+      ),
+    );
+
+    // Golden matches the backdrop filter effect of the CupertinoPopupSurface in
+    // dark mode and light mode when the popup surface is unpainted. The
+    // brightness should not affect the appearance of the popup surface.
+    await expectLater(
+      find.byType(CupertinoApp),
+      matchesGoldenFile('cupertinoPopupSurface.unpainted.filter-effects.png'),
+    );
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        theme: const CupertinoThemeData(brightness: Brightness.dark),
+        home: buildRainbowPopupSurface(isSurfacePainted: false),
+      ),
+    );
+
+    await expectLater(
+      find.byType(CupertinoApp),
+      matchesGoldenFile('cupertinoPopupSurface.unpainted.filter-effects.png'),
+    );
+  });
+}
+
+Widget buildRainbowPopupSurface({required bool isSurfacePainted}) {
+  return ColoredBox(
+    color: const Color(0xFF000000),
+    child: Builder(
+      builder: (BuildContext context) {
+        final Size size = MediaQuery.sizeOf(context);
+        final double tileHeight = size.height / 2;
+        final double tileWidth = size.width / 18;
+        return Stack(
+          fit: StackFit.expand,
+          children: <Widget>[
+            for (int i = 0; i < 18; i++)
+              Positioned(
```

Comment:
Can you explain what this designed is made for? I can see that it is a 3-layer stack, and the layer above and below the pop up surface are interlaced blocks of various colors, but why do we need it?

As far as I understand, we only need the pop up surface on top of a background that consists of various colors.

Reply:
I originally wrote it for dividers overlayed on a backdrop, so you're right, it's unnecessary. Should be fixed!